### DISCUSSION TITLE
Avatars are not loading on the Home scene for certain locales

### DISCUSSION THREAD

**Discussion by Bardreamaster** [lexiconhq/lexicon#43] [Troubleshooting and Q&A] [answered]
I'm running lexicon alpha-v2 locally and run app on iOS, with an WARN:

```
 WARN  Sending `onAnimatedValueUpdate` with no listeners registered.
```

At the same time, the avatars on main page is not available. But they are available after clicked into post page:
  
![Screenshot from 2023-08-07 17-51-52](https://github.com/lexiconhq/lexicon/assets/44565280/a7da2dca-0876-4c7d-ae12-f8e3109a9c42)
![Screenshot from 2023-08-07 17-52-06](https://github.com/lexiconhq/lexicon/assets/44565280/06cecf6c-e2be-4640-9bfb-6c2d56b254e8)
![Screenshot from 2023-08-07 17-55-31](https://github.com/lexiconhq/lexicon/assets/44565280/5130f1ae-0bc2-4b41-81bd-c01183168dc1)

Others looks good.

**Comment by gabriellaputri** [comment]
Can you provide steps to reproduce the bug?

**Reply by Bardreamaster** [reply]
Yes, just run quickstart but change discourse site to point to:

1. Download repo, change quickstart site into mine which is `forum.robook.org`. Run `npm install && npm run generate`.
2. Run `npm run quickstart`.
3. Connect front app using ExpoGo on iphone with iOS version 16.6.
4. On expo, it can be observed that :
    - A warn message: ` WARN  Sending `onAnimatedValueUpdate` with no listeners registered. `
    - The avatars on main page is not available. But they are available after clicked into post page.

**Comment by gabriellaputri** [comment]
I tried running the app and pointing it to your discourse site and noticed that the value for `authorUserId` in all posts is `null`. We use `authorUserId` to identify the author of the post and get the author's avatar. So that's why the image doesn't show up.

**Reply by Bardreamaster** [reply]
Got it. is there any configuration has influence on  `authorUserId` ? Or which config would make it not `null`? My site is using the default config. 

Thanks a lot!

**Reply by gabriellaputri** [reply, marked as answer]
After some digging, it appears that this issue is connected to how we retrieve the value for `authorUserId`. It's not directly returned from Discourse, but from our own endpoint.

```js
t.nullable.int('authorUserId', {
   resolve: ({ posters }) => {
      const author = posters.find((p: Poster) =>
         p.description.toLowerCase().includes('original poster'),
      );

      return author?.userId || author?.user?.id || null;
   },
});
```

As you can see from the code snippet above, we determine the author by checking the user's description if it includes the phrase 'original poster'. Because the phrase is in English, it didn't match the user's description on your site which uses Chinese, resulting in a null return.

Thanks for pointing this out! We'll schedule a fix on our end. And of course, feel free to jump in with any contributions if you'd like.

**Reply by Bardreamaster** [reply]
Thanks so much!

**Reply by Bardreamaster** [reply]
@gabriellaputri  I'd like to know the way of how to fix it gracefully (not just replace the phrase directly). I've checked that there are several i18n config files in discourse config in `.yml` includes these system keywords. It might be good if we search different content to get `authorUserId` for different language setting. But I've never used TypeScript before. So if you have some good idea to fix this problem from your professional view, I'll try to help to fix if I could.

**Reply by jovitazhang** [reply]
Hi @Bardreamaster, we actually need to investigate this first, since it might be a regression—it was previously working. We'll be looking into this tomorrow and ideally getting to a fix soon after. We'll keep you posted on what we've found. Thanks for your patience!

**Comment by gabriellaputri** [comment]
@Bardreamaster Just wanted to give you an update. We've fixed this issue in the latest push on the `alpha-v2` branch. Feel free to test it out and let us know how it's going. If more issues arise or you have more thoughts to share, don't hesitate to reach out again. Your input is invaluable in improving our project!
